### ISSUE TITLE
Support configuring the user environment

### ISSUE THREAD

**Issue by csadorf** [aiidalab/aiidalab-docker-stack#297]
With the adoption of the j[upyter/minimal-notebook](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html#jupyter-minimal-notebook) as the base image for this stack, the user name can be controlled via the [`NB_USER` environment variable](https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#user-related-configurations) in combination with running the container as root user.

However, this function is currently not supported in the current `edge` version, likely because of [this documented bug](https://github.com/jupyter/docker-stacks/issues/975) that was apparently fixed in more recent versions of the Jupyter stack. We are currently forced to depend on older version due to dependency conflicts.

**Comment by danielhollas** [comment]
Just noting my observation that might be relevant to this: Trying to run the full-stack image under root currently fails with:

```
/usr/local/bin/start-notebook.sh: running /usr/local/bin/before-notebook.d/20_start-postgres.sh
ERROR conda.cli.main_run:execute(33): Subprocess for 'conda run ['initdb', '-D', '/home/jovyan/.postgresql']' command failed.  (See above for error)
initdb: error: cannot be run as root
Please log in (using, e.g., "su") as the (unprivileged) user that will
own the server process.
```

**Comment by unkcpz** [comment]
I think the problem is the `NB_USER` can be set only with `--user root`. This will break up running all the startup hooks. Because the startup hooks in `before-notebook.py` are all supposed to be run as the system user rather than the root.

**Comment by danielhollas** [comment]
@unkcpz yep, that's exactly right. I provided more details in #306 if you're interested.

**Comment by unkcpz** [comment]
@danielhollas thanks! that's very helpful. But there is still no proper way to do it without hacking the `start.sh` I guess?

**Comment by danielhollas** [comment]
Yeah, I don't think there is right now. I am okay with that since I am building my own Docker image anyway.
